core/templates/domain/skill/skill-rights-backend-api.service.spec.ts

```diff
@@ -16,127 +16,103 @@
+      // The skill should not currently be cached.
+      expect(skillRightsBackendApiService.isCached('0')).toBe(false);
+      // Cache a skill.
+      skillRightsBackendApiService.cacheSkillRights('0', sampleSkillRights);
 
-    expect(successHandler).toHaveBeenCalledWith(sampleResults);
-    expect(failHandler).not.toHaveBeenCalled();
-  });
+      // It should now be cached.
+      expect(skillRightsBackendApiService.isCached('0')).toBe(true);
 
-  it('should use reject handler when fetching a skill dict from backend' +
-    ' fails', function() {
-    var successHandler = jasmine.createSpy('success');
-    var failHandler = jasmine.createSpy('fail');
+      // A new skill should not have been fetched from the backend. Also,
+      // the returned skill should match the expected skill object.
+      skillRightsBackendApiService.loadSkillRights('0').then(
+        successHandler, failHandler);
+      flushMicrotasks();
 
-    $httpBackend.expect('GET', '/skill_editor_handler/rights/0')
-      .respond(500);
-    SkillRightsBackendApiService.fetchSkillRights('0').then(
-      successHandler, failHandler);
-    $httpBackend.flush();
+      expect(successHandler).toHaveBeenCalledWith(sampleSkillRights);
+      expect(failHandler).not.toHaveBeenCalled();
+    })
+  );
 
+  it('should use reject handler when fetching a skill dict from backend' +
+  ' fails', fakeAsync(() => {
+    let successHandler = jasmine.createSpy('success');
+    let failHandler = jasmine.createSpy('fail');
+    skillRightsBackendApiService.fetchSkillRights('0').then(
+      successHandler, failHandler);
+    let req = httpTestingController.expectOne(
+      '/skill_editor_handler/rights/0');
+    req.error(new ErrorEvent('Error'));
+    flushMicrotasks();
+    expect(req.request.method).toEqual('GET');
     expect(successHandler).not.toHaveBeenCalled();
     expect(failHandler).toHaveBeenCalled();
-  });
+  }));
 
   it('should successfully fetch a skill dict from backend if it\'s not' +
-    'cached', function() {
-    var successHandler = jasmine.createSpy('success');
-    var failHandler = jasmine.createSpy('fail');
-    var sampleResults = {
-      skill_id: '0',
-      can_edit_skill_description: ''
-    };
-
-    $httpBackend.expect('GET', '/skill_editor_handler/rights/0')
-      .respond(sampleResults);
-    SkillRightsBackendApiService.loadSkillRights('0').then(
+    'cached', fakeAsync(() => {
+    let successHandler = jasmine.createSpy('success');
+    let failHandler = jasmine.createSpy('fail');
+    skillRightsBackendApiService.loadSkillRights('0').then(
       successHandler, failHandler);
-    $httpBackend.flush();
-
-    expect(successHandler).toHaveBeenCalledWith(sampleResults);
+    let req = httpTestingController.expectOne(
+      '/skill_editor_handler/rights/0');
+    req.flush(sampleSkillRights);
+    flushMicrotasks();
+    expect(skillRightsBackendApiService.isCached('0')).toBe(true);
+    expect(req.request.method).toEqual('GET');
```

Comment:
This line should be written before req.flush.

Reply:
changed it